src/main/java/com/solace/samples/TopicSubscriber.java

```diff
@@ -34,79 +37,85 @@
+        connectionFactory.setVPN(SOLACE_VPN);
+        connectionFactory.setUsername(SOLACE_USERNAME);
+        connectionFactory.setPassword(SOLACE_PASSWORD);
+        Connection connection = connectionFactory.createConnection();
 
-        // Create the subscription topic programmatically
-        final Topic topic = session.createTopic("T/GettingStarted/pubsub");
+        // Create a non-transacted, Auto ACK session.
+        Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
 
-        // Latch used for synchronizing b/w threads
-        final CountDownLatch latch = new CountDownLatch(1);
+        System.out.printf("Connected to Solace Message VPN '%s' with client username '%s'.%n", SOLACE_VPN,
+                SOLACE_USERNAME);
 
-        final MessageConsumer consumer = session.createConsumer(topic);
+        // Create the subscription topic programmatically
+        Topic topic = session.createTopic(TOPIC_NAME);
 
-        /** Anonymous inner-class for receiving messages **/
-        consumer.setMessageListener(new MessageListener() {
+        // Create the message consumer for the subscription topic
+        MessageConsumer messageConsumer = session.createConsumer(topic);
 
+        /// Use the anonymous inner class for receiving messages asynchronously
+        messageConsumer.setMessageListener(new MessageListener() {
             @Override
             public void onMessage(Message message) {
-
                 try {
                     if (message instanceof TextMessage) {
                         System.out.printf("TextMessage received: '%s'%n", ((TextMessage) message).getText());
                     } else {
                         System.out.println("Message received.");
                     }
-                    System.out.printf("Message Dump:%n%s%n", SolJmsUtility.dumpMessage(message));
-                    latch.countDown(); // unblock main thread
-                } catch (JMSException e) {
+                    System.out.printf("Message Content:%n%s%n", SolJmsUtility.dumpMessage(message));
+                    latch.countDown(); // unblock the main thread
+                } catch (JMSException ex) {
                     System.out.println("Error processing incoming message.");
-                    e.printStackTrace();
+                    ex.printStackTrace();
                 }
-
             }
         });
 
         // Start receiving messages
         connection.start();
-        System.out.println("Connected. Awaiting message...");
-
-        try {
-            latch.await(); // block here until message received, and latch will
-                           // flip
-        } catch (InterruptedException e) {
-            System.out.println("I was awoken while waiting");
-        }
-
-        // Close consumer
+        System.out.println("Awaiting message...");
+        // the main thread blocks at the next statement until a message received
+        latch.await();
```

Comment:
Mark, of course the tutorial will change and repeatedly stated it in related email threads.